### ISSUE TITLE
Experiment v14: Multi-seed ensemble training

### ISSUE THREAD

**Issue by leelasd** [mseok/PIGNet2#16]
## Problem
Single model training has inherent variance:
- Results depend on random initialization
- No way to estimate prediction uncertainty
- Production ensemble uses 4 pre-trained models

## Proposed Changes
Implement multi-seed training pipeline:
1. Train N models (default: 4) with different random seeds
2. Average predictions across models for inference
3. Optionally implement Stochastic Weight Averaging (SWA)

## Implementation
- [ ] Add `--ensemble-seeds` argument (e.g., "42,123,456,789")
- [ ] Create training loop that trains multiple models
- [ ] Save all model checkpoints with seed suffix
- [ ] Update inference to load and average multiple models
- [ ] Add checkpoint averaging option (average last K checkpoints)

## Expected Outcome
- Lower prediction variance
- More reliable uncertainty estimates
- Production-ready ensemble from single training run

## Dependencies
- Builds on #7 (v13 augmentation)

## Alternative: SWA
Instead of training multiple models, implement Stochastic Weight Averaging:
```python
swa_model = torch.optim.swa_utils.AveragedModel(model)
swa_scheduler = torch.optim.swa_utils.SWALR(optimizer, swa_lr=1e-5)
```
This averages weights during training for similar ensemble effect.
